Cabal/src/Distribution/Simple/BuildPaths.hs

```diff
@@ -170,7 +172,19 @@ getSourceFiles verbosity dirs modules = flip traverse modules $ \m -> fmap ((,)
     findFileWithExtension ["hs", "lhs", "hsig", "lhsig"] dirs (ModuleName.toFilePath m)
       >>= maybe (notFound m) (return . normalise)
   where
-    notFound module_ = die' verbosity $ "can't find source for module " ++ prettyShow module_
+    notFound :: ModuleName.ModuleName -> IO FilePath
+    notFound module_ = do
+      possibleFilePaths <- concat <$> (traverse getDirectoryContentsRecursive dirs)
+      let
+          similarFiles :: [FilePath]
+          similarFiles = map fst . List.sortBy (comparing snd) $
+                          [ (fpath, dist)
+                          | fpath <- possibleFilePaths
+                          , let fname = dropExtensions $ takeFileName fpath
+                                dist = editDistance fname $ takeFileName $ ModuleName.toFilePath module_
+                          , dist < 5 && dist < length fname]
+      die' verbosity $ "can't find source for module " ++ prettyShow module_ ++
+        "\nclosest matches for possible source files for module were " ++ intercalate ", " similarFiles
```

Comment:
`\nclosest` or `\nClosest`?
Below you went for the capitalised option (`"\nIs it…`).

Reply:
I think maybe capitalized here is best, tbh, will fix in next commit.